statechangelistener.go

```diff
@@ -27,33 +31,37 @@ type StateChangeListener struct {
 type StateID string
 
 const (
-	StateIdle        = "idle"
-	StatePoll        = "poll"
-	StateProbe       = "probe"
-	StateDownloading = "downloading"
-	StateDownloaded  = "downloaded"
-	StateInstalling  = "installing"
-	StateInstalled   = "installed"
-	StateExit        = "exit"
-	StateError       = "error"
-	StateRebooting   = "rebooting"
-)
-
-type Action string
-
-const (
-	ActionEnter = "enter"
-	ActionLeave = "leave"
+	StatePark                = "park"
+	StateEntryPoint          = "entry_point"
+	StatePoll                = "poll"
+	StateProbe               = "probe"
+	StateValidation          = "validation"
+	StateDownload            = "downloading"
+	StateInstall             = "installing"
+	StateReboot              = "rebooting"
+	StateDirectDownload      = "direct_download"
+	StatePrepareLocalInstall = "prepare_local_install"
```

Comment:
Do we trigger it for all states? I don't think we do.

Looking at the source code, it seems we handle only 3 states which does it. See:

https://github.com/UpdateHub/updatehub/blob/897157c781444558f9a0161df2846eac6fa8c615/updatehub/src/states/mod.rs#L236-L238

We need to check if previous version also triggered this when moving to error but except to this, it seems we cleaned it a lot.

Reply:
We don't trigger all that states, i'll change to handle just the 3 you mentioned.